test/e2e/report_test.go

```diff
@@ -7,88 +7,90 @@ import (
+	meteringv1 "github.com/operator-framework/operator-metering/pkg/apis/metering/v1"
 	"github.com/operator-framework/operator-metering/pkg/operator/reporting"
 	"github.com/operator-framework/operator-metering/test/reportingframework"
 )
 
 type reportProducesDataTestCase struct {
 	name          string
 	queryName     string
-	schedule      *metering.ReportSchedule
-	newReportFunc func(name, queryName string, schedule *metering.ReportSchedule, start, end *time.Time) *metering.Report
+	schedule      *meteringv1.ReportSchedule
+	newReportFunc func(name, queryName string, schedule *meteringv1.ReportSchedule, start, end *time.Time) *meteringv1.Report
 	skip          bool
 	parallel      bool
 }
 
 func testReportsProduceData(t *testing.T, testReportingFramework *reportingframework.ReportingFramework, testCases []reportProducesDataTestCase) {
-	for _, test := range testCases {
-		name := test.name
-		// Fix closure captures
-		test := test
-		t.Run(name, func(t *testing.T) {
-			if test.skip {
-				t.Skip("test configured to be skipped")
-				return
-			}
-			if test.parallel {
-				t.Parallel()
-			}
-
-			query, err := testReportingFramework.WaitForMeteringReportQuery(t, test.queryName, 5*time.Second, 5*time.Minute)
-			require.NoError(t, err, "report query for report should exist")
-
-			dsGetter := reporting.NewReportDataSourceClientGetter(testReportingFramework.MeteringClient)
-			queryGetter := reporting.NewReportQueryClientGetter(testReportingFramework.MeteringClient)
-			reportGetter := reporting.NewReportClientGetter(testReportingFramework.MeteringClient)
-
-			// get all the datasources for the query used in our report
-			dependencies, err := reporting.GetQueryDependencies(queryGetter, dsGetter, reportGetter, query, nil)
-			require.NoError(t, err, "datasources for query should exist")
-
-			require.NotEqual(t, 0, len(dependencies.ReportDataSources), "Report should have at least 1 datasource dependency")
-
-			var reportStart time.Time
-
-			// for each datasource, wait until it's EarliestImportedMetricTime is set
-			for _, ds := range dependencies.ReportDataSources {
-				_, err := testReportingFramework.WaitForMeteringReportDataSource(t, ds.Name, 5*time.Second, 5*time.Minute, func(dataSource *metering.ReportDataSource) (bool, error) {
-					if dataSource.Spec.PrometheusMetricsImporter == nil {
-						return true, nil
-					}
-					if dataSource.Status.PrometheusMetricsImportStatus != nil && dataSource.Status.PrometheusMetricsImportStatus.EarliestImportedMetricTime != nil {
-						// keep the EarliestImportedMetricTime that is the
-						// least far back, so that we ensure the reportStart is
-						// a time that all datasources have metrics for.
-						if reportStart.IsZero() || dataSource.Status.PrometheusMetricsImportStatus.EarliestImportedMetricTime.Time.After(reportStart) {
-							reportStart = dataSource.Status.PrometheusMetricsImportStatus.EarliestImportedMetricTime.Time
+	t.Run("group", func(t *testing.T) {
```

Comment:
Remove this extra sub-test closure

Reply:
Without this, the defer closure in `testInstall` isn't respected.